src/modes/thread_view/thread_view.cc

```diff
@@ -2440,34 +2440,36 @@ namespace Astroid {
         });
 
     keys.register_key ("c", "thread_view.compose_to_sender",
-        "Compose a new message to the sender of the message (or receiver if sender is you)",
+        "Compose a new message to the sender of the message (or all recipients if sender is you)",
         [&] (Key) {
           if (!edit_mode) {
-            Address _to = focused_message->sender;
-            AddressList to;
-
-            if (astroid->accounts->is_me (_to)) {
-              to = AddressList(focused_message->to ());
-              Address from = _to;
-              main_window->add_mode (new EditMessage (main_window, to.str (), from.full_address ()));
+            Address sender = focused_message->sender;
+	    Address from;
+            AddressList to, cc, bcc;
+
+	    /* Send to original sender if message is not from own account,
+	       otherwise use all recipients as in the original */
+            if (astroid->accounts->is_me (sender)) {
+              from = sender;
+              to   = AddressList(focused_message->to  ());
+              cc   = AddressList(focused_message->cc  ());
+              /* Q: can Bcc ever appear in a message received? */
```

Comment:
Yes, what I meant was, Bcc will never appear in a message that you want to compose-to-sender, except when it is a Draft, unless I am missing something. 
So copying it is a no-op most of the cases.